packages/kit/src/core/sync/create_manifest_data/index.js

```diff
@@ -172,12 +172,22 @@ export default function create_manifest_data({
 			if (item.is_page) {
 				const route = /** @type {import('types').PageData} */ (route_map.get(id));
 
-				if (item.kind === 'component') {
-					route.leaf.component = project_relative;
-
-					const { layouts, errors } = trace(tree, id, item.uses_layout, project_relative);
+				// This ensures that layouts and errors are set for pages that have no Svelte file
+				// and only redirect or throw an error, but are set to the Svelte file definition if it exists.
+				// This ensures the proper error page is used and rendered in the proper layout.
+				if (!(route.errors.length || route.layouts.length) || item.kind === 'component') {
```

Comment:
My brain can't make heads or tails of this condition — can we simplify it? Since `layouts.length` and `errors.length` will only ever be `0` at the same time, we only need to check one, I think?

```suggestion
				if (item.kind === 'component' || route.layouts.length === 0) {
```

Reply:
That's true. The root layout will always be added.